lib/endpoints/class-wp-json-comments-controller.php

```diff
@@ -329,6 +337,10 @@ public function get_item_permissions_check( $request ) {
 			return false;
 		}
 
+		if ( ! empty( $request['context'] ) && 'edit' === $request['context'] && ! current_user_can( 'moderate_comments' ) ) {
+			return new WP_Error( 'json_forbidden', __( 'Sorry, you cannot view this comment with edit context' ), array( 'status' => 403 ) );
```

Comment:
The caps check can return either. I kept the `WP_Error` so we can have a custom error message in this case.